### DISCUSSION TITLE
Welcome to nanometer Discussions!

### DISCUSSION THREAD

**Discussion by olyrhc** [olyrhc/nanometer#2] [Announcements]
## nanometer discussions

Please read the [nanometer wiki](https://github.com/olyrhc/nanometer/wiki) before you post your questions here.

You're welcome to give feedback about potential bugs you may have discovered or maybe write a few words about what you like the most about nanometer.

### Got an idea for a new feature?
Let me know!  I can't promise I will implement every suggestion, but I will at least consider it depending on how useful it is for the average user and whether or not FL Studio's API supports it.

**Comment by grantflashdance** [comment]
Hello, I love your script! It;s EXACTLY what I want for FL. Nice!

So, I had it working gloriously and seamlessly for a few days, and then it just... stopped. I made sure to have the latest KORG midi drivers, and that it was assigned within the 1-10 MIDI slots. The unit works as a general controller now but not with your script. However, I believe that at the point it stopped working, that it was pushed past MIDI slot #10, and that this may be part of the issue. I have tried all the uninstalling/reinstalling/rebooting scenarios I can think of, and it just does'nt seem to work. It does show up the MIDI settings and is marked as active, but the controller just shows the white light in the top left corner and none of the mixer channel highlighting shows up anymore. Anything I can try? Thanks again for a super well written code!

**Reply by olyrhc** [reply]
Hmm... It's hard to say exactly what the problem may be...
but I would recommend that you try to redo these steps (even if you've already done them):

1. Disconnect any other MIDI-devices you have (for now).
2. Exit FL Studio (if it's running). Start KORG KONTROL EDITOR, select your nk2 device in the editor and then go to the File-menu and open the _nanometer.nktrl2_data_ file provided with the script and click "Write Scene Data" in the Communication menu.
3. Close the editor and open FL Studio. Open the MIDI Settings from the Options menu. Make sure that there are two instances of your nk2 device in the list: "nanoKONTROL2 1 CTRL" and "nanoKONTROL2 1 SLIDER/KNOB" (if you use an older driver, both of these will just show up as "nanoKONTROL2").
4. Select one of them and then click on "Controller type" and select the nanometer script from the list.
5. Make sure that both CTRL and SLIDER/KNOB is set to the **same** port. The port-number itself should be any free port-number that's not in use by another device. Note that it's **very** important that both CTRL and SLIDER/KNOB use the **same** port-number!
6.  Click on the "Refresh device list" button and close the MIDI Settings window.

The script **should** work now...
If you for some reason have changed the MIDI Channels of your nk2 device in the KONTROL Editor, the script will not work unless you also change the MIDI Channel numbers in the _config.py_ file. By default, the channels are 1 and 14.

If it's still not working, I would try to roll back/uninstall the MIDI driver. FL Studio can actually use NK2 without the driver. Good luck, I hope you can get the script working again!

**Reply by grantflashdance** [reply]
Amazing, thank you! I think after I went above Korg's 10 MIDI device limit, FL studio changed the port numbers to not match, which I didn't remember was critical.
I've seen no real benefit to the Korg drivers and if I have much more issues I'll just revert everything to windows, which I read is about to roll out a [generic MIDI service](https://gearspace.com/board/music-computers/1434349-korg-10-midi-device-windows-driver-debacle.html) from MS that will handle all of this. 
 Thanks again!
